crates/suiop-cli/src/cli/pulumi/init.rs

```diff
@@ -34,6 +35,10 @@ impl ProjectType {
             })
             .trim()
             .to_string();
+
+        // Validate project name
+        validate_project_name(&project_name).map_err(|e| anyhow!("{}", e))?;
```

Comment:
Do you think it could be useful to catch this error and loop back around to the project name prompt? The current approach will panic on a bad name but it might be nice to just let the user try again.